lib/util/trigger.php

```diff
@@ -62,7 +62,11 @@ function parseOperand($mem)
                 if ($mem[1] == '+' || $mem[1] == '-') {
                     $count++;
                 }
-                while ($count < $max && (ctype_digit($mem[$count]) || $mem[$count] == '.')) {
+                $periodCount = 0; // Stop after reaching first period used for float
+                while ($count < $max && (ctype_digit($mem[$count]) || $periodCount < 1)) {
+                    if ($mem[$count] == '.') {
+                        $periodCount++;
+                    }
```

Comment:
This seems like it might be too greedy. As long as a period hasn't been seen, it will grab everything.

It would probably be easier to match two numeric groups:
```
                while ($count < $max && ctype_digit($mem[$count])) {
                    $count++;
                }
                if ($count < $max && $mem[$count] == '.') {
                    $count++;
                    while ($count < $max && ctype_digit($mem[$count])) {
                        $count++;
                    }
                }
```

Reply:
Ah yeah your method seems to be more efficient after running a few tests, thanks for the suggestion 👍